app/models/relationship.rb

```diff
@@ -0,0 +1,7 @@
+class Relationship < ActiveRecord::Base
+  validates :follower_id, presence: true
+  validates :leader_id, presence: true

```

Comment:
This certainly should not be here. Why are you validating Relationship? Is user inputting data here directly? You put validations only there where user has direct impact.